### PR TITLE
feat:validate proxyURL for alertManagerCfg (#6466)

### PR DESCRIPTION
**Pull request by yp969803**
## Description

fixes: #6466
added validation for proxyURL for alertManagerCfg

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification
written unit test
## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```
validate proxyURL for alertManagerCfg

```

### CODE REVIEW

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -1724,6 +1724,11 @@ func (hc *httpClientConfig) sanitize(amVersion semver.Version, logger log.Logger
 		return err
 	}
 
+	if hc.ProxyURL != "" {
+		_, err := url.Parse(hc.ProxyURL)
```
Comment (afzal442) [inline pkg/alertmanager/amcfg.go:1728]: taking the idea of validation on proxyurl obj of `oauth2` type would be better. 
https://github.com/prometheus-operator/prometheus-operator/blob/4664990a5e2f15bd1ae408d54bd77cfd90c0c7c6/pkg/alertmanager/amcfg.go#L1783-L1785

cc @simonpasquier

Comment (nicolastakashi) [reply pkg/alertmanager/amcfg.go:1728]: Sorry @afzal442 don't follow what you mean 😄 
The code snippet you shared is only about log message.
Could you clarify?

Comment (afzal442) [reply pkg/alertmanager/amcfg.go:1728]: Aha, I see. I didn't look into that much. Nice caught. Parsing was missing. I think only one extra condition to be met here is `!amVersion.GTE(semver.MustParse("0.25.0"))`. 🤔

**Review by simonpasquier — COMMENTED** [review]
we need also to add a check to convertHTTPConfig()

**Comment by yp969803** [comment]
@simonpasquier  done the above change

**Comment by simonpasquier** [comment]
Sorry reading again the code, I suggest that you add the validation [here](https://github.com/prometheus-operator/prometheus-operator/blob/35340a6c4a97a9e6899f3d0319f6c47f781c2062/pkg/apis/monitoring/v1alpha1/validation.go#L25) and [here](https://github.com/prometheus-operator/prometheus-operator/blob/35340a6c4a97a9e6899f3d0319f6c47f781c2062/pkg/apis/monitoring/v1beta1/validation.go#L25).

**Comment by simonpasquier** [comment]
Here is a short version to get you started: https://github.com/simonpasquier/prometheus-operator/commit/400c2c51a507d9b15e61d4692d161d68f750d3ae

**Pull request closed without merging**